.github/scripts/query_observe.py

```diff
@@ -0,0 +1,299 @@
+    :param stale_checks_mins: how long should difference be between received data and current timestamp
+    :param query_interval: default interval to valid from (last 30m default)
+    :return: if source is validated from current time (true, false)
+    """
+
+    AZURE_DATASET_ID = os.environ.get("AZURE_DATASET_ID")
+    OBSERVE_TOKEN_ID = os.environ.get("OBSERVE_TOKEN_ID")
+    bearer_token = get_bearer_token()
+
+    current_ts = datetime.datetime.now().timestamp()
+    current_ts_ns = int(current_ts * 1e9)
+    current_ts_string = datetime.datetime.fromtimestamp(current_ts, timezone.utc).strftime('%Y-%m-%dT%H:%M:%SZ')
+ 
+    logger.info("{}: stale_check_mins is: {} mins".format(source, stale_checks_mins))
+    logger.info("{}: query interval is: {}".format(source, query_interval))
+    logger.info("{}: Current time is: {}".format(source, current_ts_string))
+
+    pipeline = ''
+    if source == "EventHub":
+        # OPAL PIPELINE to execute on Dataset
+        pipeline = "make_col _ob_datastream_token_id:coalesce(DATASTREAM_TOKEN_ID, string(EXTRA.datastream_token_id))| " \
+                   "filter _ob_datastream_token_id = '{}'|" \
+                   "filter (string(EXTRA.source) = 'EventHub')| " \
+                   "make_col timestamp:parse_timestamp(string(FIELDS.time), 'MM/DD/YYYY HH24:MI:SS')| " \
+                   "make_col timestamp:if_null(timestamp,parse_isotime(string(FIELDS.time)))| " \
+                   "set_valid_from options(max_time_diff:30m), timestamp|" \
+                   "make_col FIELDS:parse_json(string(FIELDS))|" \
+                   "make_col source: string(EXTRA.source)|" \
+                   "make_col category:string(FIELDS.category)|" \
+                   "make_col appName:string(FIELDS.properties.appName)|" \
+                   "make_col message:string(FIELDS.properties.message)|" \
+                   "make_col time_string:string(FIELDS.time)|" \
+                   "pick_col timestamp, time_string, source, category, appName, message|" \
+                   "statsby msg_count: count_distinct(message), latest_ts: last_not_null(timestamp), group_by(source)" \
+                   "". \
+            format(OBSERVE_TOKEN_ID)
+    elif source == "ResourceManagement":
+        pipeline = "make_col _ob_datastream_token_id:coalesce(DATASTREAM_TOKEN_ID, string(EXTRA.datastream_token_id))| " \
+                   "filter _ob_datastream_token_id = '{}'|" \
+                   "filter (string(EXTRA.source) = 'ResourceManagement')|" \
+                   "make_col source: string(EXTRA.source)|" \
+                   "make_col type:string(FIELDS.type)|" \
+                   "statsby msg_count: count_distinct(type), latest_ts: last_not_null(BUNDLE_TIMESTAMP), group_by(source)" \
+                   "". \
+            format(OBSERVE_TOKEN_ID)
+    elif source == "VmMetrics":
+        pipeline = "make_col _ob_datastream_token_id:coalesce(DATASTREAM_TOKEN_ID, string(EXTRA.datastream_token_id))| " \
```

Comment:
f string f""" """ makes multiline stuff easier

Reply:
also can put variables in lin {{ }}

Reply:
I've tried 3 quotes & 2 quotes back-back but the \n new lines in 3 quotes don't play well with OPAL during query time as our API complains of invalid json

I decided to separate out the OPAL into its own file as you mentioned to make it easy to read